packages/frontend-plugin-api/src/analytics/types.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Common analytics context attributes.
+ *
+ * @public
+ */
+export type CommonAnalyticsContext = {
+  /**
+   * The nearest known parent plugin where the event was captured.
+   */
+  pluginId: string;
+
+  /**
+   * The ID of the routeRef that was active when the event was captured.
+   */
+  routeRef: string;
```

Comment:
Should we update it to use the new route references or remove it completely? The reason I'm asking is because the new type of route references has no id.

Reply:
I think we should completely move over to just using extension IDs tbh. Feels like the best starting point at the very least and we can see what feedback is on that approach

Reply:
Done https://github.com/backstage/backstage/pull/21244/commits/5c794cd5835c6c2be565134d0a744743dc95637a